test/e2e/helpers/toxiProxy.go

```diff
@@ -0,0 +1,192 @@
+package helpers
+
+import (
+	"context"
+	"fmt"
+	. "github.com/onsi/gomega"
+	"math/rand"
+	"os"
+	"path"
+	"regexp"
+	"sigs.k8s.io/cluster-api/test/framework"
+	"sigs.k8s.io/cluster-api/test/framework/exec"
+	"strconv"
+	"strings"
+)
+
+func ToxiProxyServerExec(ctx context.Context) error {
+	execArgs := []string{"run", "-d", "--name=capc-e2e-toxiproxy", "--net=host", "--rm", "ghcr.io/shopify/toxiproxy"}
+	runCmd := exec.NewCommand(
+		exec.WithCommand("docker"),
+		exec.WithArgs(execArgs...),
+	)
+	_, stderr, err := runCmd.Run(ctx)
+	if err != nil {
+		fmt.Println(string(stderr))
+	}
+	return err
+}
+
+func ToxiProxyServerKill(ctx context.Context) error {
+	execArgs := []string{"stop", "capc-e2e-toxiproxy"}
+	runCmd := exec.NewCommand(
+		exec.WithCommand("docker"),
+		exec.WithArgs(execArgs...),
+	)
+	_, _, err := runCmd.Run(ctx)
+	return err
+}
+
+func ToxiProxyCli(ctx context.Context, command string, args ...string) (string, error) {
```

Comment:
Came to see why we weren't using the ToxiProxy via an API, ouch on the package being out of date.  Worth filing an issue with their project?

Reply:
Looks like the latest is published in the v2 path, https://pkg.go.dev/github.com/Shopify/toxiproxy/v2.

Reply:
Thanks for that referenced @pmotyka.  I've re-written the methods that I used the CLI for to use the API.